### ISSUE TITLE
Quarks that are dependent on eachother cause infinite installation loop

### ISSUE THREAD

**Issue by madskjeldgaard** [supercollider/supercollider#6155]
## Environment

* SuperCollider version: 3.12
* Operating system: MacOS and Ubuntu
* Other details (Qt version, audio driver, etc.):

## Steps to reproduce

My quarks use full url's ala `https://github.com/madskjeldgaard/Monolithic` and `https://github.com/madskjeldgaard/PolyBuf` so the below example is simplified to make it clearer:

```supercollider
// Quark file in repo A
(
name: "QuarkA",
dependencies: ["QuarkB"]
);

// Quark file in repo B
(
name: "QuarkB",
dependencies: ["QuarkA"]
)
```

## Expected vs. actual behavior

I would expect dependencies to be resolved and checked off once they are installed by one of the quarks. 

Instead it goes into an infinite loop where one is causing the other to install and vice versa.

**Comment by Bubobubobubobubo** [comment]
Encountered this bug when installing `PolyBuf`.

**Comment by madskjeldgaard** [comment]
> Encountered this bug when installing `PolyBuf`. 

I think @cdbzb has fixed it in the linked PR :-)

**Comment by adcxyz** [comment]
closed with #6163
